src/borg/remote.py

```diff
@@ -57,6 +57,27 @@ def os_write(fd, data):
     return amount
 
 
+def get_limited_unpacker(kind):
+    """return a limited Unpacker because we should not trust msgpack data received from remote"""
+    args = dict(use_list=False,  # return tuples, not lists
+                max_bin_len=0,  # not used
+                max_ext_len=0,  # not used
+                max_buffer_size=3 * max(BUFSIZE, MAX_OBJECT_SIZE),
+                max_str_len=MAX_OBJECT_SIZE,  # a chunk or other repo object
+                )
+    if kind == 'server':
```

Comment:
Rather make separate args, like so:

```py
def get_limited_unpacker(*, client=False, server=False):
  i = iter((client, server))
  any(i)
  assert not any(i), "Only one arg can be True!"
```

Or if combinatorial notation is preferred:

```py
assert (a ^ b) or not (a or b), "Either client or server can be true, but not both."
```